platforms/hermes-python/hermes_python/hermes.py

```diff
@@ -0,0 +1,273 @@
+
+        :param user_callback_subscribe_session_started: the callback to be executed when a new dialogue session is started.
+        :return: the current instance of Hermes to allow chaining.
+        """
+        self._c_callback_subscribe_session_started = self._wraps(user_callback_subscribe_session_started,
+                                                                 CSessionStartedMessage, c_void_p,
+                                                                 SessionStartedMessage)
+        hermes_dialogue_subscribe_session_started(self._facade, self._c_callback_subscribe_session_started)
+        return self
+
+    def subscribe_session_queued(self, user_callback_subscribe_session_queued):
+        """
+        Register a callback when the Dialogue Manager queues the current session.
+
+        The callback will be called with the following parameters :
+            - hermes : the current instance of the Hermes object
+            - sessionQueuedMessage : message that the handler receives from the Dialogue Manager when a session is queued.
+
+        :param user_callback_subscribe_session_queued: the callback to be executed when a new dialogue session is queued.
+        :return: the current instance of Hermes to allow chaining.
+        """
+        self._c_callback_subscribe_session_queued = self._wraps(user_callback_subscribe_session_queued,
+                                                                CSessionQueuedMessage, c_void_p, SessionQueuedMessage)
+        hermes_dialogue_subscribe_session_queued(self._facade, self._c_callback_subscribe_session_queued)
+        return self
+
+    def subscribe_session_ended(self, user_callback_subscribe_session_ended):
+        """
+        Register a callback when the Dialogue Manager ends a session.
+
+        The callback will be called with the following parameters :
+            - hermes : the current instance of the Hermes object
+            - sessionEndedMessage : message that the handler receives from the Dialogue Manager when a session is ended.
+
+        :param user_callback_subscribe_session_ended: the callback to be executed when a new dialogue session is ended.
+        :return: the current instance of Hermes to allow chaining.
+        """
+        self._c_callback_subscribe_session_ended = self._wraps(user_callback_subscribe_session_ended,
+                                                               CSessionEndedMessage, c_void_p, SessionEndedMessage)
+        hermes_dialogue_subscribe_session_ended(self._facade, self._c_callback_subscribe_session_ended)
+        return self
+
+
+    def subscribe_intent_not_recognized(self, user_callback_subscribe_intent_not_recognized):
+        """
+        Register a callback when the Dialogue Manager doesn't recognize an intent.
+
+        Note that you need to have initialized a session with the intent_not_recognized field set to true.
```

Comment:
the flag can also be set on the continue session, and always concerns only the next dialogue turn